File: src/dotnet/PolyBus/Transport/Transactions/Messages/Message.cs
using System.Diagnostics;

namespace PolyBus.Transport.Transactions.Messages;

[DebuggerStepThrough]
public class Message(IPolyBus bus)
{
    /// <summary>
    /// State dictionary that can be used to store arbitrary data associated with the message.
    /// </summary>
    public virtual IDictionary<string, object> State { get; } = new Dictionary<string, object>();

    /// <summary>
    /// Message headers from the transport.
    /// </summary>
    public virtual IDictionary<string, string> Headers { get; set; } = new Dictionary<string, string>();

    /// <summary>
    /// The bus instance associated with the message.
    /// </summary>
    public IPolyBus Bus => bus ?? throw new ArgumentNullException(nameof(bus));
}

File: src/kotlin/src/main/kotlin/polybus/transport/transactions/messages/Message.kt
package polybus.transport.transactions.messages

import polybus.IPolyBus

open class Message(private val _bus: IPolyBus) {
    open val state: MutableMap<String, Any> = mutableMapOf()
    open var headers: MutableMap<String, String> = mutableMapOf()

    open val bus: IPolyBus
        get() = _bus
}
